indexinglanguage/src/main/java/com/yahoo/vespa/indexinglanguage/expressions/BinarizeExpression.java

```diff
@@ -0,0 +1,74 @@
+// Copyright Vespa.ai. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.indexinglanguage.expressions;
+
+import com.yahoo.document.DataType;
+import com.yahoo.document.DocumentType;
+import com.yahoo.document.Field;
+import com.yahoo.document.TensorDataType;
+import com.yahoo.document.datatypes.TensorFieldValue;
+import com.yahoo.tensor.Tensor;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * Converts a vector of any input type into a binarized vector.
+ *
+ * @author bratseth
+ */
+public class BinarizeExpression extends Expression  {
+
+    private final double threshold;
+
+    //private DataType targetType;
+
+    /** The type this bth consumes and produces. */
+    private DataType type;
+
+    /**
+     * Creates a binarize expression.
+     *
+     * @param threshold the value which the tensor cell value must be larger than to be set to 1 and not 0.
+     */
+    public BinarizeExpression(double threshold) {
+        super(TensorDataType.any());
+        this.threshold = threshold;
+    }
+
+    @Override
+    public void setStatementOutput(DocumentType documentType, Field field) {
+//        if (! (field.getDataType() instanceof TensorDataType))
```

Comment:
we want to pick up both the input and output tensor types and ensure they are compatible; we need to make sure we can extend this so binarize can take a `tensor<float>(x[24])` as input and produce `tensor<int8>(y[3])` as output in the future.  It looks like it might be possible to do something here, why did you comment out the check?

Reply:
My plan is to create a separate pack_bits for that.

Longer term I want to support arbitrary tensor expressions in indexing, and then we can't derive the behavior of an expression (i.e whether it should pack) from the output type.

As it is, this expression works with any tensor and does not modify the type, so there is nothing to check (I think?).